_pages/share.md

```diff
@@ -35,82 +35,123 @@ After installing the app, just click the link you were given again and choose 
-    }
-
-    if (window.location.hash) {
-        var parts = decodeURIComponent(window.location.hash.substring(1)).split("&");
-
-        // Defaults
-        var data = {
-            "store": null,
-            "cardid": null,
-            "barcodeid": null,
-            "barcodetype": null,
-            "balancetype": null,
-            "expiry": null
-        };
-
-        parts.forEach(function(part) {
-            kv = part.split("=", 2);
-            console.log(kv);
-            key = kv[0];
-            value = decodeURIComponent(kv[1].replace(/\+/g, " "));
-            console.log(key);
-            console.log(value);
-
-            data[key] = value;
-        });
-
-        innerHTML = [];
-        if (data['store'] != null) {
-            setField("Store", data["store"]);
-        };
-        if (data['barcodetype'] != null) {
-            catimaToBwipMap = {
-                "AZTEC": "azteccode",
-                "CODABAR": "rationalizedCodabar",
-                "CODE_39": "code39",
-                "CODE_93": "code93",
-                "CODE_128": "code128",
-                "DATA_MATRIX": "datamatrix",
-                "EAN_8": "ean8",
-                "EAN_13": "ean13",
-                "ITF": "interleaved2of5",
-                "MAXICODE": "maxicode",
-                "PDF_417": "pdf417",
-                "QR_CODE": "qrcode",
-                "RSS_14": "databarexpanded",
-                "RSS_EXPANDED": "databarexpanded",
-                "UPC_A": "upca",
-                "UPC_E": "upce"
-            }
-
-            try {
-                bwipjs.toCanvas(document.getElementById('sharedCardInfoBarcode'), {
-                    bcid: catimaToBwipMap[data["barcodetype"]],
-                    text: data["barcodeid"] ?? data["cardid"],
-                    includetext: false,
-                    backgroundcolor: "ffffff",
-                    padding: 2
-                });
-            } catch (e) {
-                // `e` may be a string or Error object
-            }
-        }
-
-        if (data['cardid'] != null) {
-            setField("CardID", data["cardid"]);
-        };
-        if (data['balance'] != null) {
-            setField("Balance", `${data["balance"]} ${data["balancetype"] ?? "points"}`);
-        };
-        if (data['expiry'] != null) {
-            setField("Expiry", new Date(parseInt(data["expiry"])).toLocaleDateString())
-        };
-        if (data['note'] != null) {
-            setField("Note", data["note"]); 
-        }
-    }
+function androidToRGBA (color) {
+  var colorArray = [];
+  // split android color string into it's R,G,B and A components
+  for (var i = 0; i < 4; i++) {
+    colorArray.push(color % 256);
+    color >>>= 8;
+  }
+  // remove alpha value from the original array
+  // and store it as it"s own variable
+  var alpha = colorArray.pop() / 255;
+  // return the color in RGBA format
+  return `rgba(${colorArray.reverse()},${alpha})`.replace("-", "");
+}
```

Comment:
This sadly doesn't seem to work correctly.

Take the following card as example:

![image](https://user-images.githubusercontent.com/1885159/195648957-bdb2323c-66b1-4e89-9f2c-f0dbd329eafb.png)

That generates https://catima.app/share#store%3DExample%26note%3D%26balance%3D0%26cardid%3D1234%26barcodetype%3DCODE_128%26headercolor%3D-1784274

However in your PR the colour is completely different:
![image](https://user-images.githubusercontent.com/1885159/195649114-82f1927e-6797-463b-8c62-9259f869313c.png)

I think the code may be trying to use the wrong type of Android colour, as I am pretty sure the Android code doesn't support alpha.

Reply:
mmh, that's a real bummer! I want to believe I went by the [provided documentation](https://developer.android.com/reference/android/graphics/Color) as mentions within [Catimas own reference](https://catimaloyalty.readthedocs.io/en/latest/docs/card_sharing_url_format/#parameters). The Android docs mention alpha being present; tho I might have taken a wrong turn at some point. I've switched the approach to just simply convert the given colour to HEX which seems to work quite well. Now the yellow background colour is applied properly – [Example](https://deploy-preview-19--youthful-ride-1733ce.netlify.app/share/#store%3DExample%26note%3D%26balance%3D0%26cardid%3D1234%26barcodetype%3DCODE_128%26headercolor%3D-1784274)
THe provided colour still has an Alpha value attached but since Android delivers `#AARRGGBB` and browsers read it as `#RRGGBBAA` i've just emitted that part as it's not necessary for the layout anyway. I hope that's fine.